Server/tests/test_core_infrastructure_characterization.py

```
@@ -0,0 +1,1334 @@
+# =============================================================================
+
+class TestTelemetryCollection:
+
+    @pytest.fixture(autouse=True)
+    def setup(self, fresh_telemetry):
+        """Reset telemetry before each test in this class."""
+        pass
+
+    """Tests for TelemetryCollector basic functionality."""
+
+    def test_telemetry_collector_initialization(self, mock_telemetry_config, temp_telemetry_data):
```

Ruff (0.14.14)

Unused method argument: `mock_telemetry_config`

(ARG002)